internal/controller/compaction/reconciler.go

```diff
@@ -359,6 +353,129 @@ func (r *Reconciler) createCompactionJob(ctx context.Context, logger logr.Logger
 	return job, nil
 }
 
+func handleFailedJob(ctx context.Context, r *Reconciler, logger logr.Logger, job *batchv1.Job, etcd *druidv1alpha1.Etcd, jobCompletionReason string) error {
+	switch jobCompletionReason {
+	case batchv1.JobReasonDeadlineExceeded:
+		logger.Info("Job has been completed due to deadline exceeded", "namespace", job.Namespace, "name", job.Name)
+		metricJobsTotal.With(prometheus.Labels{druidmetrics.LabelSucceeded: druidmetrics.ValueSucceededDeadlineExceeded, druidmetrics.EtcdNamespace: etcd.Namespace}).Inc()
+		metricJobDurationSeconds.With(prometheus.Labels{druidmetrics.LabelSucceeded: druidmetrics.ValueSucceededDeadlineExceeded, druidmetrics.EtcdNamespace: etcd.Namespace}).Observe(float64(*job.Spec.ActiveDeadlineSeconds))
+	case batchv1.JobReasonBackoffLimitExceeded:
+		if err := handleBackoffLimitExceeded(ctx, r, logger, job, etcd); err != nil {
```

Comment:
This is the parent method where except in one case of `batchv1.JobReasonBackoffLimitExceeded` in other cases the metrics are recorded in the same method. It becomes easy if you can compute the metrics values elsewhere but record it here instead of doing that finally in `handlePodFailureReason`